proof-systems/src/darlin/poly-commit/src/ipa_pc/mod.rs

```diff
@@ -192,6 +199,79 @@ impl<G: IPACurve, FS: FiatShamirRng> InnerProductArgPC<G, FS> {
         )
     }
 
+    #[cfg(feature="circuit-friendly")]
+    pub fn open_reduction_polynomials<'a>(
+        ck: &CommitterKey<G>,
+        xi_s: impl IntoIterator<Item = &'a LabeledSuccinctCheckPolynomial<'a, G>>,
+        point: G::ScalarField,
+        // Assumption: the evaluation point and the (xi_s, g_fins) are already bound to the
+        // fs_rng state.
+        fs_rng: &mut FS,
+    ) -> Result<Proof<G>, Error> {
+        let mut key_len = ck.comm_key.len();
+        if ck.comm_key.len().next_power_of_two() != key_len {
+            return Err(Error::Other(
+                "Commiter key length is not power of 2".to_owned(),
+            ));
+        }
+        let batch_time = start_timer!(|| "Compute and batch Bullet Polys and GFin commitments");
+
+        let mut xi_s_vec = xi_s.into_iter().collect::<Vec<_>>();
+
+        // Compute the evaluations of the Bullet polynomials at point starting from the xi_s
+        let values = xi_s_vec
+            .par_iter()
+            .map(|xi_s| xi_s.get_poly().evaluate(point))
+            .collect::<Vec<_>>();
+        // record evaluations
+        fs_rng.record(values)?;
+
+        // sort in reverse lexicographical order according to labels
+        xi_s_vec.sort_by(|xi_1, xi_2| xi_2.get_label().cmp(xi_1.get_label()));
```

Comment:
Can we put under the circuit-friendly flag only this instruction, instead of duplicating the whole open_reduction_polynomials function ?

Reply:
Ok